### DISCUSSION TITLE
Whats the fastest way of refreshing clangd with a modified compile_commands.json?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [clangd/clangd#750] [Q&A]
Im currently calling clangd.restart (in VS Code) which makes clangd just re-index everything seemingly, are there any other alternatives for having clangd read changes made to a compile_commands.json?

**Comment by HighCommander4** [comment]
A couple of notes:

 * clangd.restart should not cause clangd to re-index everything. It will cause clangd to re-read what's already been indexed from disk, which is perhaps what you're seeing.
 * clangd has [recently gained](https://github.com/clangd/clangd/issues/83) the ability to automatically watch for changes to `compile_commands.json` and reload the file when it detects changes. I believe this has made it into the recent clangd 12 release.

**Reply by ItsCubeTime** [reply]
Ah, excuse me for my misasssumption, I will see if I can get that automatic reload working instead <3

Thanks

**Reply by ItsCubeTime** [reply]
Doesnt seem to be in the latest Github releases neither on LLVM/releases nor on clangd/releases, Im guessing there are no newer binaries anywhere?

**Reply by njames93** [reply]
https://github.com/clangd/clangd/releases/tag/12.0.0

**Reply by ItsCubeTime** [reply]
Yeah I tried that one as well as https://github.com/clangd/clangd/releases/tag/snapshot_20210418

**Reply by kadircet** [reply]
> Doesnt seem to be in the latest Github releases neither on LLVM/releases nor on clangd/releases, Im guessing there are no newer binaries anywhere?

They should contain the hot-reload feature. What is not working for you?

**Reply by ItsCubeTime** [reply]
> > Doesnt seem to be in the latest Github releases neither on LLVM/releases nor on clangd/releases, Im guessing there are no newer binaries anywhere?
> 
> They should contain the hot-reload feature. What is not working for you?

Yeah I saw the commit linked by HighCommander was tagged with rc13, I just went by

>  I believe this has made it into the recent clangd 12 release.

**Reply by kirillbobyrev** [reply]
The latest snapshot should for sure include the needed patch (https://github.com/llvm/llvm-project/commit/98993193e9037345ad13720a62974064a5f3d953) but so is clangd 12.0.0.

The changed aren't picked up momentarily but they should appear after some time. The reload happens when the `compile_commands.json` are being used again and clangd understands it's stale and it's also time to reload it. You could try opening some file, doing some operations, then changing `compile_commands.json`, waiting some time and doing some operations that would reveal whether the `compile_commands.json` are updated again. If this isn't working for you, this is surely a bug but I think this should indeed work.
